### ISSUE TITLE
get recipes from references

### ISSUE THREAD

**Issue by Tumerskiy** [Tumerskiy/BIGsortRW#2]

**Comment by Tumerskiy** [comment]
https://stackoverflow.com/questions/19486077/java-fastest-way-to-read-through-text-file-with-2-million-lines
so no magic -  BufferedReader.readLine() for reading.

https://stackoverflow.com/questions/1062113/fastest-way-to-write-huge-data-in-text-file-java
here as suggested we can play around with FileWriter with hope that we will land to disk's cache.
Trickiest thing here we have to well aware of the hardware config to properly align tuples to block size and be aware of disk caching capabilities.

**Comment by Mandeepahlawat** [comment]
@Tumerskiy you can't use  BufferedReader because there is no separator in between columns, only way to detect they are of different columns is by no of bytes so we'll have to read it by a different method using byte class or something. There are solutions online but again I didn't understand :P  

For writing I think we can use buffered reader as we don't have to keep check on bytes

**Comment by Tumerskiy** [comment]
Do we have sample instances of T1 and T2? I think depends on the row structure.
we have to conduct couple of small practical experiments with these methods.

**Comment by Mandeepahlawat** [comment]
Yeah you have sample of relations in the project document itself. It's like this:

12345678John Smiths 111222999999999 1455 Maisonneuve West,
Montreal, QC, H3G 1M8
88888888Roselyn Shiri 222444888888888 1515 Saint-Catherine West,
Montreal, QC, H3G 1M7 

and even that is not fixed so we have to use the byte size mentioned in the schema only

**Comment by Tumerskiy** [comment]
https://docs.oracle.com/javase/tutorial/essential/io/file.html
let's go through our options

**Comment by Mandeepahlawat** [comment]
I think the bytesbuffer? what do you say? Ideally I was thinking if we could just read the whole line and then just read bytes from that line but I don't think that's possible, right?

**Comment by PhanKhang** [comment]
You can just use method readline. In java we have substring and trim, so its ok to have some spaces. Then convert it to int or long for comparator
